src/test/java/net/syscon/elite/repository/jpa/repository/OffenderNonAssociationRepositoryTest.java

```diff
@@ -0,0 +1,30 @@
+package net.syscon.elite.repository.jpa.repository;
+
+import net.syscon.elite.security.AuthenticationFacade;
+import net.syscon.elite.web.config.AuditorAwareImpl;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ActiveProfiles;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace.NONE;
+
+@DataJpaTest
+@ActiveProfiles("test")
+@AutoConfigureTestDatabase(replace = NONE)
+@Import({AuthenticationFacade.class, AuditorAwareImpl.class})
+@WithMockUser
+public class OffenderNonAssociationRepositoryTest {
+
+    @Autowired
+    private OffenderNonAssociationRepository nonAssociationRepository;
+
+    @Test
+    void find_non_associations_by_offender_booking() {
+        assertThat(nonAssociationRepository.findAllByOffenderBooking_BookingId(-1L)).hasSize(1);
```

Comment:
Is it work testing that the complex properties are properly hydrated? Some of the joins could be valid SQL but might be joining on the wrong properties? Be good to have a test at least from a regression standpoint.